src/main/java/com/gargoylesoftware/htmlunit/util/KeyDataPair.java

```diff
@@ -159,4 +160,25 @@ private void readObject(final ObjectInputStream ois) throws ClassNotFoundExcepti
             charset_ = Charset.forName(charsetName);
         }
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * specialization of inherited method which will copy all fields
+     * and make sure that the value in the base class is not null, by calling
+     * the constructor with the current value
+     */
+    @Override
+    public KeyDataPair normalized() {
+        final KeyDataPair pair = new KeyDataPair(
+            this.getName(),
+            this.fileObject_ != null ? this.fileObject_ : new File(this.getValue()),
```

Comment:
new File(...) might do some normalization on the file name leading in the end to some new value -> Testcase?
maybe it is better to have a private constructor that allows to pass the values 1:1?

What do you think?